Sources/SwiftSyntaxBuilder/SyntaxNodeWithBody.swift

```diff
@@ -39,16 +39,31 @@ extension SyntaxStringInterpolation {
 
 public protocol HasTrailingCodeBlock {
   var body: CodeBlockSyntax { get set }
+
+  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) throws
 }
 
-public extension HasTrailingCodeBlock where Self: SyntaxExpressibleByStringInterpolation {
-  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) {
-    self = "\(signature) {}"
+public extension HasTrailingCodeBlock where Self: StmtSyntaxProtocol {
+  init(_ signature: PartialSyntaxNodeString, @CodeBlockItemListBuilder bodyBuilder: () -> CodeBlockItemListSyntax) throws {
+    let stmt = StmtSyntax("\(signature) {}")
+    guard let castedStmt = stmt.as(Self.self) else {
+      throw SyntaxStringInterpolationError.producedInvalidNodeType(expectedType: IfStmtSyntax.self, actualNode: stmt)
+    }
+    self = castedStmt
     self.body = CodeBlockSyntax(statements: bodyBuilder())
   }
 }
```

Comment:
Still not a huge fan of this one, but if you really want it `expectedType: IfStmtSyntax.self` is likely not what you intended :P. `Self.self`? Also odd that we use `StmtSyntax`, I assume that's because the majority of these are statements. Could split it into `StmtHasTrailing...` and `DeclHasTrailing...`?

Reply:
> Still not a huge fan of this one, but if you really want it `expectedType: IfStmtSyntax.self` is likely not what you intended :P. `Self.self`? 

Oh, absolutely. Yes. Again, copy-paste error from the `IfStmt` below.

Also odd that we use `StmtSyntax`, I assume that's because the majority of these are statements. Could split it into `StmtHasTrailing...` and `DeclHasTrailing...`?